twingate/internal/provider/datasource/groups.go

```diff
@@ -8,83 +8,102 @@ import (
 
-	groups, err := c.ReadGroups(ctx, filter)
-	if err != nil && !errors.Is(err, client.ErrGraphqlResultIsEmpty) {
-		return diag.FromErr(err)
-	}
+func NewGroupsDatasource() datasource.DataSource {
+	return &groups{}
+}
 
-	if err := resourceData.Set(attr.Groups, convertGroupsToTerraform(groups)); err != nil {
-		return diag.FromErr(err)
-	}
+type groups struct {
+	client *client.Client
+}
 
-	id := "all-groups"
-	if filter.HasName() {
-		id = "groups-by-name-" + *filter.Name
+type groupsModel struct {
+	ID       types.String `tfsdk:"id"`
+	Name     types.String `tfsdk:"name"`
+	Type     types.String `tfsdk:"type"`
+	IsActive types.Bool   `tfsdk:"is_active"`
+	Groups   []groupModel `tfsdk:"groups"`
+}
+
+func (d *groups) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = TwingateGroups
+}
+
+func (d *groups) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
 	}
 
-	resourceData.SetId(id)
+	client, ok := req.ProviderData.(*client.Client)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *client.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
 
-	return nil
+	d.client = client
 }
 
-func Groups() *schema.Resource {
-	return &schema.Resource{
+func (d *groups) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
 		Description: "Groups are how users are authorized to access Resources. For more information, see Twingate's [documentation](https://docs.twingate.com/docs/groups).",
-		ReadContext: datasourceGroupsRead,
-		Schema: map[string]*schema.Schema{
-			attr.Name: {
-				Type:        schema.TypeString,
+		Attributes: map[string]schema.Attribute{
+			attr.ID: schema.StringAttribute{
+				Computed:    true,
+				Description: computedDatasourceIDDescription,
+			},
+			attr.Name: schema.StringAttribute{
 				Optional:    true,
 				Description: "Returns only Groups that exactly match this name.",
 			},
-			attr.IsActive: {
-				Type:        schema.TypeBool,
+			attr.IsActive: schema.BoolAttribute{
 				Optional:    true,
 				Description: "Returns only Groups matching the specified state.",
 			},
-			attr.Type: {
-				Type:         schema.TypeString,
-				Optional:     true,
-				Description:  fmt.Sprintf("Returns only Groups of the specified type (valid: `%s`, `%s`, `%s`).", model.GroupTypeManual, model.GroupTypeSynced, model.GroupTypeSystem),
-				ValidateFunc: validation.StringInSlice([]string{model.GroupTypeManual, model.GroupTypeSynced, model.GroupTypeSystem}, false),
+			attr.Type: schema.StringAttribute{
+				Optional:    true,
+				Description: fmt.Sprintf("Returns only Groups of the specified type (valid options: %s).", utils.DocList(model.GroupTypes)),
```

Comment:
Why did you make this change?   this doesnt seem related to the move to new sdk

Reply:
it's just to simplify docs updates, I've notice that we using in multiple places such list of enums
so when we need to add new entry we can only change list at one place